project/app/views.py

```diff
@@ -72,20 +114,64 @@ def post(self, request):
 
 
 class AdminAccessView(ListAPIView, CreateAPIView):
-    permission_classes = [IsAuthenticated,RequiredAdmin]
+    permission_classes = [IsAuthenticated, RequiredAdmin]
 
     queryset = UserProfile.objects.all()
     serializer_class = UserProfileSerializer
 
 
 class UpdateProfileView(APIView):
-    permission_classes = [IsAuthenticated,RequiredAdmin]
+    permission_classes = [IsAuthenticated, RequiredAdmin]
 
-    def put(self, request,pk):
+    def put(self, request, pk):
         user_profile = UserProfile.objects.get(user_id=pk)
         profile_serializer = UserProfileSerializer(instance=user_profile, data=request.data)
 
         if not profile_serializer.is_valid():
             return Response(status=status.HTTP_404_NOT_FOUND)
         profile_serializer.save()
         return Response(profile_serializer.data)
+
+
+class AdminAddUserView(APIView):
+    permission_classes = [IsAuthenticated, RequiredAdmin, AdminCanCreate]
+
+    def post(self, request):
+        serializer = UserSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer.save()
+        email = request.POST.get("email")
+        user = CustomUser.objects.get(email=email)
+        user.is_active = True
+        user.save()
+        return Response(serializer.data)
+
+
+class ManagerAddUserView(APIView):
+    permission_classes = [IsAuthenticated, RequiredManager, ManagerCanCreate]
```

Comment:
No need for Two different views for Admin and Manager to create a user.
As Both have same functionality.
So, If it's manager required then admin can do it as well

Reply:
but there is a difference that admin can add both employee and manager but manager only can add employee

Reply:
For that you can check if the role of user is admin, then get the manage-id from request body. And if the role is manager then take the manager id from request.user.id